src/app/components/project-map/new-template-dialog/new-template-dialog.component.ts

```diff
@@ -0,0 +1,156 @@
+    ],
+})
+export class NewTemplateDialogComponent implements OnInit {
+    @Input() server: Server;
+    @Input() project: Project;
+
+    uploader: FileUploader;
+
+    public action: string = 'install';
+    public actionTitle: string = 'Install appliance from server';
+
+    public searchText: string = '';
+    public allAppliances: Appliance[] = [];
+    public appliances: Appliance[] = [];
+
+    public categories: string[] = ['all categories', 'router', 'multilayer_switch', 'guest', 'firewall'];
+    public category: string = 'all categories';
+    public displayedColumns: string[] = ['name', 'emulator', 'vendor'];
+
+    public dataSource: MatTableDataSource<Appliance>;
+
+    @ViewChild(MatPaginator, {static: true}) paginator: MatPaginator;
+
+    constructor(
+        public dialogRef: MatDialogRef<NewTemplateDialogComponent>,
+        private applianceService: ApplianceService,
+        private changeDetector: ChangeDetectorRef,
+        private toasterService: ToasterService
+    ) {}
+
+    ngOnInit() {
+        this.applianceService.getAppliances(this.server).subscribe((appliances) => {
+            this.appliances = appliances;
+            this.appliances.forEach(appliance => {
+                if (appliance.docker) appliance.emulator = 'Docker';
+                if (appliance.dynamips) appliance.emulator = 'Dynamips';
+                if (appliance.iou) appliance.emulator = 'Iou';
+                if (appliance.qemu) appliance.emulator = 'Qemu';
+            });
+            this.allAppliances = appliances;
+            this.dataSource = new MatTableDataSource(this.allAppliances);
+            this.dataSource.paginator = this.paginator;
+        });
+
+        this.uploader =  new FileUploader({});
+        this.uploader.onAfterAddingFile = file => {
+            file.withCredentials = false;
+        };
+    
+        this.uploader.onErrorItem = (item: FileItem, response: string, status: number, headers: ParsedResponseHeaders) => {
+            this.toasterService.error('An error has occured');
+        };
+        
+        this.uploader.onSuccessItem = (item: FileItem, response: string, status: number, headers: ParsedResponseHeaders) => {
+            this.toasterService.success('Appliance imported succesfully');
+        };
+    }
+
+    addAppliance(event): void {
+        let name = event.target.files[0].name.split('-')[0];
+        let fileName = event.target.files[0].name;
+        let file = event.target.files[0];
+        let fileReader: FileReader = new FileReader();
+        let emulator;
+
+        fileReader.onloadend = () => {
+            let appliance = JSON.parse(fileReader.result as string);
+
+            if (appliance.docker) emulator = 'docker';
+            if (appliance.dynamips) emulator = 'dynamips';
+            if (appliance.iou) emulator = 'iou';
+            if (appliance.qemu) emulator = 'qemu';
+
+            const url = this.applianceService.getUploadPath(this.server, emulator, fileName);
```

Comment:
@ziajka still an error occurs during import

Reply:
any updates?

Reply:
Indeed, there is an error when importing a .gns3a file.

![Screenshot from 2020-06-11 16-37-33](https://user-images.githubusercontent.com/4752614/84355517-e8dd8d80-ac01-11ea-9c20-040a3f160b05.png)

Unfortunately, I don't see why you get this error... :(

Reply:
@grossmj Can u confirm that this endpoint `http://${server.host}:${server.port}/v2/${emulator}/images/${filename}` is correct?

Reply:
Looks like you are missing `/compute` after `/v2`.

Here is a request example:

```
curl -v -H "Content-Type:application/octet-stream" "http://localhost:3080/v2/compute/dynamips/images/c7200-adventerprisek9-mz.153-3.XB12.bin" --data-binary "@c7200-adventerprisek9-mz.153-3.XB12.bin"
```